Added selectors for the Order Activity Log, that merges order notes with label events

DanReyLop opened the pull request:
This PR adds selectors to merge Order Notes and Shipping Label-related events, so they can be consumed by the same list in the frontend.

There's no code using this yet, so to test, just check that the tests pass.

dechov commented on client/extensions/woocommerce/state/sites/orders/activity-log/selectors.js:
```diff
@@ -0,0 +1,95 @@
+	} ) );
+
+	if ( config.isEnabled( 'woocommerce/extension-wcservices' ) ) {
+		const labels = get( state, [ 'extensions', 'woocommerce', 'woocommerceServices', siteId, 'shippingLabel', orderId, 'labels' ], [] );
```
I feel like we should we use the [WCS selector](https://github.com/Automattic/wp-calypso/pull/17676/files#diff-111b3d32a30c9153b2b5a2bf4fb01035R10) for the shipping label state — what do you think? (FYI there's also one for the [labels themselves](https://github.com/Automattic/wp-calypso/pull/17757/files#diff-111b3d32a30c9153b2b5a2bf4fb01035R29) that I added in #17757)

DanReyLop replied:
Good! I'll steal the selector you added on #17757. You may have some trouble rebasing though :(

dechov replied:
Nope— no conflict

DanReyLop commented:
This is ready for review now.

dechov commented on client/extensions/woocommerce/state/sites/orders/activity-log/selectors.js:
```diff
@@ -0,0 +1,143 @@
+				key: label.label_id,
+				type: EVENT_TYPES.LABEL_PURCHASED,
+				timestamp: label.created_date,
+				labelIndex: index,
```
The labels have been numbered as `labels.length - index` to this point https://github.com/Automattic/woocommerce-services/blob/a8974547f3867ebcad0015b865bb76e558266538/client/apps/shipping-label/view.js#L98
Do we want to stick with that?

DanReyLop replied:
Yes we do, great catch! The server replies with the last label first, so we must flip the index here.
Done in https://github.com/Automattic/wp-calypso/pull/17807/commits/1cce3554757dafd4f81168d35b795f456149c162

DanReyLop commented on client/extensions/woocommerce/state/sites/orders/activity-log/test/selectors.js:
```diff
@@ -0,0 +1,424 @@
+		} );
+
+		it( 'should be false when notes are loaded but labels are not.', () => {
+			expect( isActivityLogLoaded( labelsLoadingState, 45, 123 ) ).to.be.falsy;
```
Because the function returns `something_that_evaluates_to_null() && something_that_also_evaluates_to_null()`, which is `null`. The only important thing is that it returns "true-ish" or "falsy" values in a boolean context, so I just changed it in the cases where it was failing.

dechov commented:
Ready to merge AFAICT but keep in mind https://github.com/Automattic/wp-calypso/pull/17757#issuecomment-333180482

DanReyLop commented:
> Ready to merge AFAICT but keep in mind #17757 (comment)

I added the same properties in https://github.com/Automattic/wp-calypso/pull/17807/commits/ad57cdbb6bc8e3f64075786d4c041466d609e8a3 and updated the tests accordingly.
